includes/class-wc-payments.php

```diff
@@ -603,8 +602,13 @@ public static function possibly_add_source_to_notes_query( $args, $request ) {
 	 * @return string Modified where clause.
 	 */
 	public static function possibly_add_note_source_where_clause( $where_clauses, $args ) {
-		if ( isset( $args['source'] ) && false === strpos( $where_clauses, 'AND source IN' ) ) {
-			$escaped_where_source = sprintf( "'%s'", esc_sql( $args['source'] ) );
+		if ( isset( $args['source'] ) && ! empty( $args['source'] ) && false === strpos( $where_clauses, 'AND source IN' ) ) {
```

Comment:
Nitpick: it's probably redundant to have `isset` and `! empty` together.  I think you could simplify it like:
```suggestion
		if ( ! empty( $args['source'] ) && false === strpos( $where_clauses, 'AND source IN' ) ) {
```